docs/developers/packaging.md

```diff
@@ -312,6 +309,125 @@ a high-level list of the main changes introduced in the stack.
 * Add signing for Windows
 * Add notarization for macOS PKG
 
+#### `constructor-manager` stack for handling updates and recovery
+
+Constructor installers come with a `constructor-manager` stack that allows users to perform
+"updates in place", recover from a broken installation, rollback to a previous version or restore
+point and finally uninstall the entire application. The constructor manager stack is application
+agnostic in that it can be used for any othe application using the updated constructor stack
+to build and distribute installers that may or may not handle additional plugins.
+
+![Constructor manager stack](images/constructor_manager_stack.png)
+
+This stack is composed of three main packages [19]:
+
+* `constructor-manager-backend`: Provides a simplified command line interface on top of
+  `conda`, `mamba` and `conda-lock` to query for updates and perform the actual installation,
+  reset or uninstallation of the application.
+* `constructor-manager-client`: Provides a Python API to interact with the backend and
+  perform the same operations as the CLI.
```

Comment:
Hi @aganders3 yes indeed this might be confusing. Let me explain and from there we can make adjustments

The backend right now is only a CLI, that under the hood calls mamba, conda, conda-lock, and does a bunch of admin work for us (update in place, restore points etc). This could be updated eventually to become something more of a cli, for example an actual server doing things. So in this case instead of pinging a local server, we ping an actual program CLI.

The client is what we use to communicate with this backend (currently a cli but maybe a server in the future). So that is why I went with that naming scheme. The client could be renamed API. I called it like that in the past actually. But in this case is one thing we use to communicate and do thins with the "server" (like a frontend backend thing).

But I am open to suggestions, not super important but I would like to get the naming "right" before publishing packages

Reply:
I will go with `constructor-manager` and `constructor-manager-api`, that way we leave the cli to be more open and add a specific entrypoint in there `constructor-manager-cli` and leave the api one as the api that it is :) 

Thanks for the feedback